dbt/adapters/snowflake/relation.py

```diff
@@ -33,3 +51,48 @@ def is_dynamic_table(self) -> bool:
     @classproperty
     def DynamicTable(cls) -> str:
         return str(SnowflakeRelationType.DynamicTable)
+
+    @classproperty
+    def get_relation_type(cls) -> Type[SnowflakeRelationType]:
+        return SnowflakeRelationType
+
+    @classmethod
+    def from_runtime_config(cls, runtime_config: RuntimeConfigObject) -> RelationConfigBase:
```

Comment:
What does this function mean? I tried figuring that out from reading where it is invoked. That said, it's not immediately clear. You want the model from the runtime config which the materialization invokes? Just a lot of indirection here.

Reply:
I definitely need a docstring here; I'll add it. The idea is to produce a `RelationConfig` from the `RuntimeConfigObject`; in this case it would produce a `SnowflakeDynamicTableConfig` instance. `RuntimeConfigObject` contains all of the information needed to create a `SnowflakeDynamicTableConfig`, plus a lot more. There are two key differences. The latter is in the parlance of the Snowflake object, and not the generic `RuntimeConfigObject`. And producing the latter provides a much better place for validation checks and structural checks; we don't need to do it at the `SnowflakeRelation` level or in the jinja.